Sources/MapboxCoreNavigation/Router.swift

```diff
@@ -404,10 +404,18 @@ extension InternalRouter where Self: Router {
 extension Array where Element: MapboxDirections.Route {
     func index(mostSimilarTo route: Route) -> Int? {
         let target = route.description
-        return enumerated().min { (left, right) -> Bool in
-            let leftDistance = left.element.description.minimumEditDistance(to: target)
-            let rightDistance = right.element.description.minimumEditDistance(to: target)
-            return leftDistance < rightDistance
-        }?.offset
+        
+        guard let bestCandidate = map({
+            (route: $0, editDistance: $0.description.minimumEditDistance(to: target))
+        }).enumerated().min(by: { $0.element.editDistance < $1.element.editDistance }) else { return nil }
+
+
+        // If the most similar route is still more than 50% different from the original route,
+        // we fallback to the fastest route which index is 0.
+        guard bestCandidate.element.route.description.count > 0 else { return 0 }
+        let similarityScore = Double(bestCandidate.element.editDistance) / Double(bestCandidate.element.route.description.count)
+        guard similarityScore < 0.5 else { return 0 }
```

Comment:
Thanks, I inverted this logic in https://github.com/mapbox/mapbox-navigation-ios/issues/3597#issuecomment-982362158 because I mistook `score` for a measure of similarity, but in fact it’s still based on the edit distance, so it’s really a relative difference.